ios-target-app/Home/ViewModels/HomeViewModel.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  HomeViewModel.swift
+//  ios-target-app
+//
+//  Created by Raul Piñeres Carrera on 18/10/22.
+//
+
+import Foundation
+import Combine
+import CoreLocation
+
+final class HomeViewModel {
+  
+  private var cancellables: Set<AnyCancellable> = []
+  
+  private let locationManager: LocationManager
+  
+  init(locationManager: LocationManager = LocationManager.instance) {
+    self.locationManager = locationManager
+    
+    setupSubscribers()
+  }
+  
+  @Published private var currentLocation: CLLocation = CLLocation()
+  var currentLocationPublisher: Published<CLLocation>.Publisher { $currentLocation }
```

Comment:
do you really need the currentLocation variable?
You can directly pass the signal from the location manager:
```suggestion
  var currentLocationPublisher: Published<CLLocation>.Publisher { 
    locationManager.$lastLocation
  }
```

Reply:
gotcha